src/main/java/org/folio/circulation/domain/CheckInContext.java

```diff
@@ -237,4 +241,23 @@ public ItemStatus getItemStatusBeforeCheckIn() {
   public boolean areLostItemFeesRefundedOrCancelled() {
     return lostItemFeesRefundedOrCancelled;
   }
+
+  public JsonObject asJson() {
+    JsonObject json = new JsonObject();
+    write(json, "item", JsonObject.mapFrom(item));
+    write(json, "loan", loan.asJsonWithUserAndProxy());
+    write(json, "checkInRequest", JsonObject.mapFrom(checkInRequest));
+    write(json, "requestQueue", JsonObject.mapFrom(requestQueue));
+    write(json, "checkInServicePoint", JsonObject.mapFrom(checkInServicePoint));
+    if (Objects.nonNull(highestPriorityFulfillableRequest)) {
+      write(json, "highestPriorityFulfillableRequest", highestPriorityFulfillableRequest.asJson());
+    }
+    write(json, "loggedInUserId", loggedInUserId);
+    write(json, "checkInProcessedDateTime", checkInProcessedDateTime);
+    write(json, "inHouseUse", inHouseUse);
+    writeByPath(json, itemStatusBeforeCheckIn.getDate(), "itemStatusBeforeCheckIn", "date");
+    writeByPath(json, itemStatusBeforeCheckIn.getValue(), "itemStatusBeforeCheckIn", "value");
+    write(json, "lostItemFeesRefundedOrCancelled", lostItemFeesRefundedOrCancelled);
+    return json;
+  }
```

Comment:
Why does this context need encoding as JSON?

Reply:
The answer is here https://github.com/folio-org/mod-circulation/pull/683#discussion_r493052737. I just can mention here that we can reduce content of this json when we will be sure that it really contains unnecessary information. But only after log record processors in the mod-audit will be completed. Because now I'm not sure what particular info we need to extract from it. Maybe we will face with requirements that we need to extend log records or something else.

Reply:
> I just can mention here that we can reduce content of this json when we will be sure that it really contains unnecessary information. But only after log record processors in the mod-audit will be completed. Because now I'm not sure what particular info we need to extract from it. Maybe we will face with requirements that we need to extend log records or something else.

I think it is better to start with publishing only the data that is known to be be wanted by the circulation log and add to it if that is insufficient. Even though it is implicit at the moment, the messages sent to the audit form a versioned contract. It is often possible to add information without breaking compatibility whereas removing it because we didn't need it is naturally compatibility breaking.